IdentityCore/src/cache/crypto/MSIDSymmetricKey.m

```diff
@@ -0,0 +1,177 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "MSIDSymmetricKey.h"
+
+#import <CommonCrypto/CommonCryptor.h>
+#import <CommonCrypto/CommonHMAC.h>
+#import <CommonCrypto/CommonDigest.h>
+
+@implementation MSIDSymmetricKey
+
+- (nullable instancetype)initWithSymmetericKeyBytes:(NSData *)symmetericKey {
+    if (!symmetericKey)
+    {
+        return nil;
+    }
+    
+    self = [super init];
+    
+    if (self)
+    {
+        _symmetericKey = symmetericKey;
+    }
+    
+    return self;
+}
+
+- (nullable instancetype)initWithSymmetericKeyBase64:(NSString *)symmetericKeyBase64 {
+    if (!symmetericKeyBase64)
+    {
+        return nil;
+    }
+    
+    return [self initWithSymmetericKeyBytes:[[NSData alloc] initWithBase64EncodedString:symmetericKeyBase64 options:0]];
+}
+
+- (nullable NSString *)createVerifySignature:(NSData *)context
+                                  dataToSign:(NSString *)dataToSign {
+    NSData *data = [dataToSign dataUsingEncoding:NSUTF8StringEncoding];
+    NSData *derivedKey = [self computeKDFInCounterMode:context];
+    unsigned char cHMAC[CC_SHA256_DIGEST_LENGTH];
+    CCHmac(kCCHmacAlgSHA256,
+           derivedKey.bytes,
+           derivedKey.length,
+           [data bytes],
+           [data length],
+           cHMAC);
+    NSData *signedData = [[NSData alloc] initWithBytes:cHMAC length:sizeof(cHMAC)];
+
+    return [NSString msidBase64UrlEncodedStringFromData:signedData];
+}
+
+- (NSData *)computeKDFInCounterMode:(NSData *)ctx
+{
+    NSData *labelData = [@"AzureAD-SecureConversation" dataUsingEncoding:NSUTF8StringEncoding];
+    NSMutableData *mutData = [NSMutableData new];
+    [mutData appendBytes:labelData.bytes length:labelData.length];
+    Byte bytes[] = {0x00};
+    [mutData appendBytes:bytes length:1];
+    [mutData appendBytes:ctx.bytes length:ctx.length];
+    int32_t size = CFSwapInt32HostToBig(256); //make big-endian
+    [mutData appendBytes:&size length:sizeof(size)];
+    
+    uint8_t *pbDerivedKey = [self KDFCounterMode:(uint8_t*)_symmetericKey.bytes
+                          keyDerivationKeyLength:_symmetericKey.length
+                                      fixedInput:(uint8_t*)mutData.bytes
+                                fixedInputLength:mutData.length];
+    mutData = nil;
+    NSData *dataToReturn = [NSData dataWithBytes:(const void *)pbDerivedKey length:32];
+    free(pbDerivedKey);
+    
+    return dataToReturn;
+}
+
+
+- (uint8_t*) KDFCounterMode:(uint8_t*) keyDerivationKey
```

Comment:
nit codestyle: remove space after closing curly brace and start method name with lowercase

So something like:

- (uint8_t *)kdfCounterMode:(uint8_t *)keyDerivationKey